src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: string): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  items.forEach((item: any) => {
```

Comment:
Avoid using `any` in TypeScript, it's a bad practice. Consider setting the correct type or just remove the typization here, as `items` is an array of T and it's clear what element of the array is